crates/core/tedge_agent/src/file_transfer_server/http_rest.rs

```diff
@@ -54,327 +68,370 @@ impl HttpConfig {
             ..self
         }
     }
-
-    pub fn file_transfer_end_point(&self) -> String {
-        format!("{}file-transfer/*", self.file_transfer_uri)
-    }
-
-    /// Return the path of the file associated to the given `uri`
-    ///
-    /// Check that:
-    /// * the `uri` is related to the file-transfer i.e a sub-uri of `self.file_transfer_uri`
-    /// * the `path`, once normalized, is actually under `self.file_transfer_dir`
-    pub fn local_path_for_uri(&self, uri: String) -> Result<Utf8PathBuf, FileTransferError> {
-        let ref_uri = uri.clone();
-
-        // The file transfer prefix has to be removed from the uri to get the target path
-        let path = uri
-            .strip_prefix(&self.file_transfer_uri)
-            .ok_or(FileTransferError::InvalidURI { value: ref_uri })?;
-
-        // This path is relative to the file transfer dir
-        let full_path = self.data_dir.join(path);
-
-        // One must check that once normalized (i.e. any `..` removed)
-        // the path is still under the file transfer dir
-        let clean_path = clean_utf8_path(&full_path);
-        if clean_path.starts_with(&self.data_dir) {
```

Comment:
This was only checking that the file was in e.g. `/var/tedge` since all the manipulation in this function was based around `file-transfer` being part of the URL path (i.e. in the variable `path` in this function), so it was possible for a user to read from/write to `/var/tedge`, not just `/var/tedge/file-transfer`. I'm assuming this wasn't intentional and I've fixed it by storing `file_transfer_dir` inside `HttpConfig` instead.